src/vrapp/VolumeVisualizationApp.cpp

```diff
@@ -493,25 +493,46 @@ void VolumeVisualizationApp::onGenericEvent(const VRDataIndex& index)
   if (index.getName() == "WindowClose") {
     shutdown();
   }
+
+  if (index.getName() == "WindowSize") {
+    /*
+    some code to update projection matrix
+    */
+  }
+
+  if (index.getName() == "BufferSize") {
+    const std::vector<float>* v = index.getValue("WindowSize");
+    const float* f = &(v->front());
+    std::cout << "new buffer size: " << GLint(f[0]) << ", " << GLint(f[1]) << std::endl;
+    glViewport(0, 0, GLint(f[0]), GLint(f[1]));
+  }
+
 }
 
 void VolumeVisualizationApp::onRenderGraphicsContext(const VRGraphicsState& renderState) {
   // This routine is called once per graphics context at the start of the
   // rendering process.  So, this is the place to initialize textures,
   // load models, or do other operations that you only want to do once per
   // frame when in stereo mode.
+  m_num_frames++;
 
   std::chrono::steady_clock::time_point nowTime = std::chrono::steady_clock::now();
-  std::chrono::duration<double> time_span = std::chrono::duration<double>(nowTime - m_lastTime);
-  float fps = 1000.0f / std::chrono::duration_cast<std::chrono::milliseconds>(time_span).count();
+  std::chrono::duration<double> delta_time = std::chrono::duration<double>(nowTime - m_lastTime);
+  float delta_time_to_milliseconds = std::chrono::duration_cast<std::chrono::milliseconds>(delta_time).count();
+  float fps = 1000.0f / delta_time_to_milliseconds;
+  
+  std::chrono::duration<double> time_lapse = std::chrono::duration<double>(nowTime - last_Update_Time);
```

Comment:
what's the difference between `delta_time` and `time_lapse`?

Reply:
I am taking out the time_lapse logic. It is left over code to try to handle a fixed refresh time rate, but I couldnt make it work properly.

Reply:
https://github.com/brown-ccv/VR-Volumeviewer/issues/50